saw-central/src/SAWCentral/Crucible/MIR/TypeShape.hs

```diff
@@ -385,27 +386,28 @@ fieldShapeMirTy (OptField shp) = shapeMirTy shp
 -- type.  For example, if a Cryptol function expects [3][8], we could map
 -- it to a Rust function that either expects `[u8;3]`, or `&[u8]` with a
 -- dynamic check that it has 3 elements.
-data CryTermAdaptor =
-    NoAdapt                      -- ^ Use default translation
-  | AdaptTuple [CryTermAdaptor]  -- ^ Adapt a tuple
-  | AdaptArray CryTermAdaptor    -- ^ Adapt an array
-  | AdaptDerefSlice Integer
+data CryTermAdaptor a =
```

Comment:
Please document what the `a` type parameter is meant to represent. I _think_ it's only ever used to represent the length of a slice, but there are situations where `a` needs to be different types (e.g., `Integer` or `Cry.Type`) depending on context.